### PR TITLE
Return invites as https, various URL normalization

### PR DESCRIPTION
**Pull request by Gorialis**
### Summary

This pull request:
- Makes invites use `https://discord.gg/` instead of `http://discord.gg/`
- Normalizes URLs for the CDN in `Asset` using a `BASE` url (like Route)
- Normalizes URLs for invites using a `BASE` url (like Route)
- Fixes a minor documentation mistake related to `GuildChannel.overwrites`

This changes the behavior of Asset to natively refer to URLs under the CDN instead of absolute URLs like they used to be. `str(asset)` will still give you the full URL as normal, and constructors have been adjusted, but `Asset._url` and the repr will now show the URL relative to the root of the domain.

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/abc.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def overwrites(self):
         """Returns all of the channel's overwrites.
 
         This is returned as a dictionary where the key contains the target which
-        can be either a :class:`~discord.Role` or a :class:`~discord.Member` and the key is the
+        can be either a :class:`~discord.Role` or a :class:`~discord.Member` and the value is the
```
Comment (iomintz) [inline discord/abc.py]: This seems like an unrelated change. Separate PR?

File: discord/asset.py [hunk 2]
```diff
@@ -132,15 +134,15 @@ def _from_guild_icon(cls, state, guild, *, format=None, static_format='webp', si
         if format is None:
             format = 'gif' if guild.is_icon_animated() else static_format
 
-        return cls(state, 'https://cdn.discordapp.com/icons/{0.id}/{0.icon}.{1}?size={2}'.format(guild, format, size))
+        return cls(state, '/icons/{0.id}/{0.icon}.{1}?size={2}'.format(guild, format, size))
 
 
     def __str__(self):
-        return self._url if self._url is not None else ''
+        return self.BASE + self._url if self._url is not None else ''
```
Comment (apple502j) [inline discord/asset.py]: is it really `self.BASE`?

Comment (iomintz) [reply discord/asset.py]: https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables
